pkg/deployment/deployment.go

```diff
@@ -121,15 +129,28 @@ func Deploy(ctx context.Context, helper *helper.Helper, obj client.Object, sshKe
 
 // ConditionalDeploy function encapsulating primary deloyment handling with
 // conditions.
-func ConditionalDeploy(ctx context.Context, helper *helper.Helper, obj client.Object, sshKeySecret string, inventoryConfigMap string, status *dataplanev1beta1.OpenStackDataPlaneNodeStatus, readyCondition condition.Type, readyMessage string, readyWaitingMessage string, deployFunc deployFuncDef, deployName string, deployLabel string) (ctrl.Result, error) {
+func ConditionalDeploy(
+	ctx context.Context,
+	helper *helper.Helper,
+	obj client.Object,
+	sshKeySecret string,
+	inventoryConfigMap string,
+	status *dataplanev1beta1.OpenStackDataPlaneNodeStatus,
+	readyCondition condition.Type,
+	readyMessage string,
+	readyWaitingMessage string,
+	deployFunc deployFuncDef,
+	deployName string,
+	deployLabel string,
+	networkAttachments []string,
```

Comment:
we might want to introduce a custom struct type to pass information instead of that many params